mitodl/mitxpro
# Integrated Wagtail to support course/program detail pages

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #155 

#### What's this PR do?
Properly integrates Wagtail so that Wagtail serves course/program detail pages

#### How should this be manually tested?
- Run migrations
- Check that the CMS has the correct page hierarchy and that "View Live" links for detail pages work correctly
- Unapply migrations and check the CMS still looks correct (i.e.: no course/program index pages, detail pages are once again children of the top level page)

#### Any background context you want to provide?
- `cms/templates/detail_page.html` and the partial templates in the same directory are basically copies of the same templates that were deleted in the `courses` app
- I added logic to exclude `Course`s/`Program`s from the catalog page that don't have an associated Wagtail page. This is something I discussed in person with Ferdi and it makes sense as a requirement

#### Screenshots (if appropriate)
![ss 2019-05-13 at 12 09 22 ](https://user-images.githubusercontent.com/14932219/57637154-af5ace80-7578-11e9-8c23-bcc6bdbe515a.png)
![ss 2019-05-13 at 12 09 46 ](https://user-images.githubusercontent.com/14932219/57637162-b1249200-7578-11e9-9a7c-cfebb3d586af.png)
